bee-tangle/src/tangle.rs

```diff
@@ -1,759 +1,133 @@
-// Copyright 2020 IOTA Stiftung
-//
-// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
-// the License. You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
-// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and limitations under the License.
+use crate::{vertex::Vertex, TransactionRef as TxRef};
```

Comment:
Out of curiosity, why not use `TransactionRef ` ?

Reply:
Same reason why we use `Btrit` and `Utrit` instead of `BalancedTrit` and `UnbalancedTrit`, only with the difference is that I very rarely like to shorten the type names themselves as this can create confusion or not self-explanatory code (requires comments to understand the intent!). I do see the need of brevity though, as if you have a long type name, and it's used repeatedly like `TransactionRef`, the redundancy of characters can make the code harder to read as method signatures, closures, etc are more likely to being "verticalized" by the formatter.

Reply:
The difference with `BTrit` and `UTrit` is that `BalancedTrit` and `UnbalancedTrit` don't exist in the first place. I usually don't like abbreviations and here is no difference. IMO `TxRef` is harder to read than `TransactionRef`. Just needed to say it :)

Reply:
Fully get your point. And I generally agree with you on this. I am up for a vote on this. There are pros and cons to this. I wouldn't do this for uncommon type names, but since `Transaction` is commonly - although falsely - abbreviated as `Tx` throughout the ecosystem I don't think this poses much of a readability issue to the wider community. It allows me in many cases to create much shorter signatures, so they stay in one line. I don't know about you, but I find multiline parameter lists of function signatures extremely ugly, and hard to read. Anyways, I am willing to change it back if need be.

That `BalancedTrit` and `UnbalancedTrit` don't exist in the first place is a mistake in my opinion. I would always define the type fully spelled out to be self-explanatory, and not having to rely on the documentation to explain the abbreviation.

Reply:
I mean such a "always". There are good cases where this is bollocks. See `Arc`.